urbit/sur/eth-event-viewer.hoon

```diff
@@ -0,0 +1,16 @@
+/-  *eth-watcher
+|%
++$  contract-type
+  $:  address=@ux
+      name=@t
+      specific-events=(list @t)
+      abi-events=@t
```

Comment:
You could consider storing the more-specific types here, instead of plain cords. ie, the `specific-events` ultimately get converted to `(list ?(@ux (list @ux)))`, so why not store that instead?

I see these do get passed onto the frontend as-is, which you can't do as trivially with complex nouns. So maybe they're just easier to work with this way?

Reply:
yes It was mainly because it is easier to handle the events, I would keep it as it is although you really want me to do transform it. I've already tried to decode the abi-events but weren't able to do so.. So I would need your help to do that.